ddht/rpc_handlers.py

```diff
@@ -0,0 +1,92 @@
+from typing import Iterator, Mapping, Tuple, TypedDict
+
+from eth_enr import ENRAPI
+from eth_typing import HexStr
+from eth_utils import encode_hex, to_dict
+
+from ddht.abc import RoutingTableAPI, RPCHandlerAPI, RPCRequest
+from ddht.rpc import RPCError, RPCHandler
+
+
+class BucketInfo(TypedDict):
+    idx: int
+    nodes: Tuple[HexStr, ...]
+    replacement_cache: Tuple[HexStr, ...]
+    is_full: bool
+
+
+class TableInfoResponse(TypedDict):
+    center_node_id: HexStr
+    num_buckets: int
+    bucket_size: int
+    buckets: Mapping[int, BucketInfo]
+
+
+class RoutingTableInfoHandler(RPCHandler[None, TableInfoResponse]):
+    def __init__(self, routing_table: RoutingTableAPI) -> None:
+        self._routing_table = routing_table
+
+    def extract_params(self, request: RPCRequest) -> None:
+        if request.get("params"):
+            raise RPCError(f"Unexpected RPC params: {request['params']}",)
+        return None
+
+    async def do_call(self, params: None) -> TableInfoResponse:
+        stats = TableInfoResponse(
+            center_node_id=encode_hex(self._routing_table.center_node_id),
+            num_buckets=len(self._routing_table.buckets),
+            bucket_size=self._routing_table.bucket_size,
+            buckets=self._bucket_stats(),
+        )
+        return stats
+
+    @to_dict
+    def _bucket_stats(self) -> Iterator[Tuple[int, BucketInfo]]:
+        buckets_and_replacement_caches = zip(
+            self._routing_table.buckets, self._routing_table.replacement_caches,
+        )
+        for idx, (bucket, replacement_cache) in enumerate(
+            buckets_and_replacement_caches, 1
```

Comment:
*nit*: I had to look up what this parameter is for, passing `start=1` is slightly more clear